### PR TITLE
feat: update Optimism and Base information for the Cancun update

### PR DESCRIPTION
**Pull request by dikel**
Optimism support all Cancun EIPs except 4844. Also `BLOBBASEFEE` always returns 1. ([source](https://gov.optimism.io/t/upgrade-proposal-5-ecotone-network-upgrade/7669))

### CODE REVIEW

**Review by Daniel-K-Ivanov — CHANGES_REQUESTED** [review]
It seems that there are 2 new system contracts - - https://specs.optimism.io/protocol/predeploys.html#gaspriceoracle and https://specs.optimism.io/protocol/deposits.html#l1-attributes---ecotone

https://eips.ethereum.org/EIPS/eip-4788 is supported on OP/Base. This means that for all rollup that do not support the EIP we need to add a system contract which they do not support. EIP4788 isa system contract introduced on the Layer 1 and L2s may or may not support it. We should document it on all rollups

**Comment by dikel** [comment]
Both system contracts are already added to the website. Additionally I checked the backtesting results of the [Delta upgrade](https://gov.optimism.io/t/final-upgrade-proposal-3-delta-network-upgrade/7310) and it doesn't lower the gas cost for L2->L1 messaging

File: src/docs/base.mdx [hunk 1]
```diff
@@ -112,34 +112,45 @@ links:
...
+    | <Copy value="0x420000000000000000000000000000000000001a" label="0x420...01a" /> | <Reference label="L1FeeVault" url="https://github.com/ethereum-optimism/optimism/blob/master/packages/contracts-bedrock/contracts/L2/L1FeeVault.sol" /> | Receives the L1 portion of the transaction fees. Once the contract has received a certain amount of fees, the ETH can be withdrawn to an immutable address on L1 | N/A <Added /> |
+    | <Copy value="0x000F3df6D732807Ef1319fB7B8bB8522d0Beac02" label="0x000...c02" /> | <Reference label="BeaconRootsStorage" url="https://eips.ethereum.org/EIPS/eip-4788" /> | The contract has the following differences compared to the L1: <br /><br /> 1. It is 7 slots behind the L1 contract <br /> 2. It only stores the hashes for ~4.5 instead of ~26 hours | The contract receives a timestamp and returns the beacon root associated with it <Modified /> |
```
Comment (Daniel-K-Ivanov) [inline src/docs/base.mdx:153]:
```suggestion
    | <Copy value="0x000F3df6D732807Ef1319fB7B8bB8522d0Beac02" label="0x000...c02" /> | <Reference label="BeaconRootsStorage" url="https://eips.ethereum.org/EIPS/eip-4788" /> | The contract has the following differences compared to the L1: <br /><br /> 1. Data is updated with ~6-7 slots of delay compared to the L1 contract <br /> 2. `HISTORY_BUFFER_LENGTH ` is set to 1024 compared to 8191, meaning that the history is for ~3.41 hours compared to ~27 hours on L1 | The contract receives a timestamp and returns the beacon root associated with it <Modified /> |
```

File: src/docs/optimism.mdx [hunk 2]
```diff
@@ -110,34 +110,45 @@ links:
...
+    | <Copy value="0x420000000000000000000000000000000000001a" label="0x420...01a" /> | <Reference label="L1FeeVault" url="https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/L2/L1FeeVault.sol" /> | Receives the L1 portion of the transaction fees. Once the contract has received a certain amount of fees, the ETH can be withdrawn to an immutable address on L1 | N/A <Added /> |
+    | <Copy value="0x000F3df6D732807Ef1319fB7B8bB8522d0Beac02" label="0x000...c02" /> | <Reference label="BeaconRootsStorage" url="https://eips.ethereum.org/EIPS/eip-4788" /> | The contract has the following differences compared to the L1: <br /><br /> 1. It is 7 slots behind the L1 contract <br /> 2. It only stores the hashes for ~4.5 instead of ~26 hours | The contract receives a timestamp and returns the beacon root associated with it <Modified /> |
```
Comment (Daniel-K-Ivanov) [inline src/docs/optimism.mdx:151]:
```suggestion
 | <Copy value="0x000F3df6D732807Ef1319fB7B8bB8522d0Beac02" label="0x000...c02" /> | <Reference label="BeaconRootsStorage" url="https://eips.ethereum.org/EIPS/eip-4788" /> | The contract has the following differences compared to the L1: <br /><br /> 1. Data is updated with ~6-7 slots of delay <br /> 2. `HISTORY_BUFFER_LENGTH ` is set to 1024 compared to 8191, meaning that the history is for ~3.41 hours compared to ~27 hours on L1 | The contract receives a timestamp and returns the beacon root associated with it <Modified /> |
```

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Pull request merged**